runelite-client/src/main/java/net/runelite/client/plugins/microbot/zerozero/tormenteddemons/TormentedDemonScript.java

```diff
@@ -473,6 +476,89 @@ private void teleportToFeroxEnclave() {
         logOnceToChat("No Ring of Dueling found in inventory for teleporting to Ferox Enclave.");
     }
 
+    private void evaluateAndConsumePotions(TormentedDemonConfig config) {
+        int currentStrength = Microbot.getClient().getBoostedSkillLevel(Skill.STRENGTH);
+        int maxStrength = Microbot.getClient().getRealSkillLevel(Skill.STRENGTH);
+        int currentRange = Microbot.getClient().getBoostedSkillLevel(Skill.RANGED);
+        int maxRange = Microbot.getClient().getRealSkillLevel(Skill.RANGED);
+
+        int threshold = config.boostedStatsThreshold();
+
+        List<Integer> combatPotionIds = null;
+        switch (config.combatPotionType()) {
+            case SUPER_COMBAT:
+                combatPotionIds = List.of(
+                        ItemID.SUPER_COMBAT_POTION4,
+                        ItemID.SUPER_COMBAT_POTION3,
+                        ItemID.SUPER_COMBAT_POTION2,
+                        ItemID.SUPER_COMBAT_POTION1
+                );
+                break;
+            case DIVINE_SUPER_COMBAT:
+                combatPotionIds = List.of(
+                        ItemID.DIVINE_SUPER_COMBAT_POTION4,
+                        ItemID.DIVINE_SUPER_COMBAT_POTION3,
+                        ItemID.DIVINE_SUPER_COMBAT_POTION2,
+                        ItemID.DIVINE_SUPER_COMBAT_POTION1
+                );
+                break;
+        }
+
+        List<Integer> rangingPotionIds = null;
+        switch (config.rangingPotionType()) {
+            case RANGING:
+                rangingPotionIds = List.of(
```

Comment:
this is missing normal bastion potion. we have a utility function called: 

`Rs2Potion.getRangePotionsVariants()`

You can use it as:

boolean hasRangePotion = `Rs2Inventory.hasItem(Rs2Potion.getRangePotionsVariants())`

or interact with:

`interact(Rs2Potion.getRangePotionsVariants(), ""Drink")`